staging/src/k8s.io/apiserver/pkg/endpoints/handlers/namer.go

```diff
@@ -100,15 +100,6 @@ func (n ContextBasedNaming) GenerateLink(req *http.Request, obj runtime.Object)
 		return "", fmt.Errorf("missing requestInfo")
 	}
 
-	if len(namespace) == 0 && len(name) == 0 {
-		if len(requestInfo.Name) == 0 {
```

Comment:
@xiangpengzhao @soltysh 
This is the new method:
If ObjectName returns empty string or errEmptyName, we refactor to requestInfo.